From f364766e1f58fe5109e97105bf71c8003db823ec Mon Sep 17 00:00:00 2001
From: Guillem Jover <guillem@debian.org>
Date: Thu, 3 Jul 2014 01:56:52 +0200
Subject: [PATCH 08/11] build: Set the XSL parameter through the command line
 instead of sed

Use the xsltproc --stringparam option instead of replacing it inside the
file.
---
 buildlib/po4a_manpage.mak | 7 ++++---
 doc/manpage-style.xsl     | 1 -
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/buildlib/po4a_manpage.mak b/buildlib/po4a_manpage.mak
index b3d586b..62ada19 100644
--- a/buildlib/po4a_manpage.mak
+++ b/buildlib/po4a_manpage.mak
@@ -35,12 +35,13 @@ apt-vendor.ent: ../apt-vendor.ent
 	cp -a ../apt-vendor.ent .
 
 manpage-style.xsl: ../manpage-style.xsl
-	sed "/<!-- LANGUAGE -->/ i\
-<xsl:param name=\"l10n.gentext.default.language\" select=\"'$(LC)'\" />" ../manpage-style.xsl > manpage-style.xsl
+	cp -a $< .
 
 $($(LOCAL)-LIST) :: % : %.xml $(STYLESHEET) $(INCLUDES)
 	echo Creating man page $@
-	$(XSLTPROC) -o $@ $(STYLESHEET) $< || exit 200 # why xsltproc doesn't respect the -o flag here???
+	$(XSLTPROC) \
+		--stringparam l10n.gentext.default.language $(LC) \
+		-o $@ $(STYLESHEET) $< || exit 200 # why xsltproc doesn't respect the -o flag here???
 	test -f $(subst .$(LC),,$@) || echo 'FIXME: xsltproc respects the -o flag now, workaround can be removed'
 	mv -f $(subst .$(LC),,$@) $@
 
diff --git a/doc/manpage-style.xsl b/doc/manpage-style.xsl
index 5cb6a58..a780bad 100644
--- a/doc/manpage-style.xsl
+++ b/doc/manpage-style.xsl
@@ -5,7 +5,6 @@
 <xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/manpages/docbook.xsl" />
 
 <xsl:param name="man.output.encoding" select="'UTF-8'" />
-<!-- LANGUAGE -->
 
 <xsl:template match="email">&lt;<xsl:apply-templates/>&gt;</xsl:template>
 
-- 
2.0.1

